### ISSUE TITLE
Docs: no-unsafe-finally used in generator functions

### ISSUE THREAD

**Issue by TomStrepsil** [eslint/eslint#20313]
### Docs page(s)

https://eslint.org/docs/latest/rules/no-unsafe-finally

### What documentation issue do you want to solve?

[no-unsafe-finally](https://eslint.org/docs/latest/rules/no-unsafe-finally) warns for a canonical use case (as [described in MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Generator/return)) for use in generator functions:
```js
function* gen() {
  try {
    yield 1;
  } finally {
    return "cleanup";
  }
}
```

### What do you think is the correct solution?

The ["When not to use it"](https://eslint.org/docs/latest/rules/no-unsafe-finally) section could perhaps make it clear that this is a use-case with specific semantics.

### Participation

- [x] I am willing to submit a pull request for this change.

**Comment by DMartens** [comment]
I think rather than only changing the documentation, we should support this case in the rule.
You can go ahead and file a PR for the documentation change.
Then depending whether the team wants to support this case, I can update the rule.

**Comment by TomStrepsil** [comment]
OK - thanks, I will claim this issue (for documentation) and will work on this now, should be relatively quick I'd hope.
